src/commands/deploy.js

```diff
@@ -16,20 +16,263 @@ const SitesCreateCommand = require('./sites/create')
 const LinkCommand = require('./link')
 const { NETLIFYDEV, NETLIFYDEVLOG, NETLIFYDEVERR } = require('../utils/logo')
 
+const triggerDeploy = async ({ api, siteId, siteData, log, error }) => {
+  try {
+    const siteBuild = await api.createSiteBuild({ siteId })
+    log(
+      `${NETLIFYDEV} A new deployment was triggered successfully. Visit https://app.netlify.com/sites/${siteData.name}/deploys/${siteBuild.deploy_id} to see the logs.`
+    )
+  } catch (err) {
+    if (err.status === 404) {
+      error('Site not found. Please rerun "netlify link" and make sure that your site has CI configured.')
+    } else {
+      error(err.message)
+    }
+  }
+}
+
+const getDeployFolder = async ({ flags, config, site, siteData, log }) => {
+  let deployFolder
+  if (flags['dir']) {
+    deployFolder = path.resolve(process.cwd(), flags['dir'])
+  } else if (get(config, 'build.publish')) {
+    deployFolder = path.resolve(site.root, get(config, 'build.publish'))
+  } else if (get(siteData, 'build_settings.dir')) {
+    deployFolder = path.resolve(site.root, get(siteData, 'build_settings.dir'))
+  }
+
+  if (!deployFolder) {
+    log('Please provide a publish directory (e.g. "public" or "dist" or "."):')
+    log(process.cwd())
+    const { promptPath } = await inquirer.prompt([
+      {
+        type: 'input',
+        name: 'promptPath',
+        message: 'Publish directory',
+        default: '.',
+        filter: input => path.resolve(process.cwd(), input),
+      },
+    ])
+    deployFolder = promptPath
+  }
+
+  return deployFolder
+}
+
+const validateDeployFolder = async ({ deployFolder, error }) => {
+  let stat
+  try {
+    stat = await fs.stat(deployFolder)
+  } catch (e) {
+    if (e.code === 'ENOENT') {
+      return error(`No such directory ${deployFolder}! Did you forget to run a build?`)
+    }
+
+    // Improve the message of permission errors
+    if (e.code === 'EACCES') {
+      return error('Permission error when trying to access deploy folder')
+    }
+    throw e
+  }
+
+  if (!stat.isDirectory()) {
+    return error('Deploy target must be a path to a directory')
+  }
+  return stat
+}
+
+const getFunctionsFolder = ({ flags, config, site, siteData }) => {
+  let functionsFolder
+  // Support "functions" and "Functions"
+  const funcConfig = get(config, 'build.functions') || get(config, 'build.Functions')
+  if (flags['functions']) {
+    functionsFolder = path.resolve(process.cwd(), flags['functions'])
+  } else if (funcConfig) {
+    functionsFolder = path.resolve(site.root, funcConfig)
+  } else if (get(siteData, 'build_settings.functions_dir')) {
+    functionsFolder = path.resolve(site.root, get(siteData, 'build_settings.functions_dir'))
+  }
+  return functionsFolder
+}
+
+const validateFunctionsFolder = async ({ functionsFolder, log, error }) => {
```

Comment:
Extracted from https://github.com/netlify/cli/blob/39da8081eaca6e2c8ba58aa782541199e0df48fc/src/commands/deploy.js#L114 and fixed to reference `stat.isDirectory()`